libs/framework/include/celix/dm/ServiceDependency.h

```diff
@@ -219,13 +229,13 @@ namespace celix { namespace dm {
     class ServiceDependency : public TypedServiceDependency<T> {
         using type = I;
     public:
-        ServiceDependency(const std::string &name = std::string{}, bool valid = true);
+        ServiceDependency(celix_dm_component_t* cCmp, const std::string &name, bool valid = true);
         ~ServiceDependency() override = default;
 
         ServiceDependency(const ServiceDependency&) = delete;
         ServiceDependency& operator=(const ServiceDependency&) = delete;
-        ServiceDependency(ServiceDependency&&) noexcept = default;
-        ServiceDependency& operator=(ServiceDependency&&) noexcept = default;
+        ServiceDependency(ServiceDependency&&) noexcept = delete;
+        ServiceDependency& operator=(ServiceDependency&&) noexcept = delete;
```

Comment:
It's late so maybe I'm missing something, but semantically, I would expect moving of all these service dependencies to be OK. What's the reason you put it on delete?

Reply:
the base of service dependency has raw pointer, to prevent compiler issues the move and copy constructions are deleted